AnkiDroid/src/main/java/com/ichi2/anki/multiprofile/ProfileManager.kt

```diff
@@ -0,0 +1,254 @@
+        Timber.i("No active profile found. Setting up Default.")
+        val defaultId = ProfileId(DEFAULT_PROFILE_ID)
+
+        val metadata =
+            ProfileMetadata(
+                displayName = DEFAULT_PROFILE_DISPLAY_NAME,
+                isLegacy = true,
+            )
+
+        profileRegistry.saveProfile(defaultId, metadata)
+        profileRegistry.setLastActiveProfileId(defaultId)
+
+        return defaultId
+    }
+
+    fun createNewProfile(displayName: String): ProfileId {
+        val newProfileId = generateUniqueProfileId()
+
+        val metadata =
+            ProfileMetadata(
+                displayName = displayName,
+                isLegacy = false,
+            )
+
+        profileRegistry.saveProfile(newProfileId, metadata)
+
+        Timber.i("Created new profile: $displayName (${newProfileId.value})")
+        return newProfileId
+    }
+
+    private fun generateUniqueProfileId(): ProfileId {
+        var newId: ProfileId
+        var collisionCount = 0
+        do {
+            if (collisionCount > 0) Timber.w("Profile ID collision detected!")
+            val rawId = "p_" + UUID.randomUUID().toString().substring(0, 8)
+            newId = ProfileId(rawId)
+            collisionCount++
+        } while (profileRegistry.contains(newId))
+        return newId
+    }
+
+    fun switchActiveProfile(newProfileId: ProfileId) {
+        Timber.i("Switching profile to ID: ${newProfileId.value}")
+
+        profileRegistry.setLastActiveProfileId(newProfileId)
+        triggerAppRestart()
+    }
+
+    private fun loadProfileData(profileId: ProfileId) {
+        configureWebView(profileId)
+
+        val profileBaseDir = resolveProfileDirectory(profileId)
+
+        try {
+            activeProfileContext =
+                ProfileContextWrapper.create(
+                    context = appContext,
+                    profileId = profileId,
+                    profileBaseDir = profileBaseDir,
+                )
+        } catch (e: Exception) {
+            Timber.e(e, "Failed to load profile context for ${profileId.value}")
+            throw RuntimeException("Failed to load profile environment", e)
+        }
+
+        Timber.d("Profile loaded: ${profileId.value} at ${profileBaseDir.absolutePath}")
+    }
+
+    private fun configureWebView(profileId: ProfileId) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+            CookieManager.getInstance().removeAllCookies(null)
+            return
+        }
+
+        if (profileId.isDefault()) {
+            return
+        }
+
+        try {
+            WebView.setDataDirectorySuffix(profileId.value)
+        } catch (e: Exception) {
+            // This usually means the WebView was accessed before the profile was loaded.
+            // This represents a potential privacy leak (using default cookies).
+            Timber.e(e, "Failed to set WebView directory suffix (WebView already initialized?)")
```

Comment:
All of the `e` should be `w`

```suggestion
            Timber.w(e, "Failed to set WebView directory suffix (WebView already initialized?)")
```